.github/workflows/courses.yml

```diff
@@ -71,9 +71,18 @@ jobs:
         git add _data/courses.yml
         git commit -m "Automatically update courses.yml"
         git push origin master
+    - uses: fabasoad/yaml-json-xml-converter-action@v1.0.0
+      id: yaml2json
+      with:
+        path: bi/_data/courses.yml
+        from: 'yaml'
+        to: 'json'
+        
+    - name: Print JSON to file
+      run: echo '${{ steps.yaml2json.outputs.data }}' > courses.json
         
     - name: Upload courses to Spaces
-      run: aws s3 --endpoint cp ams3.digitaloceanspaces.com _data/courses.yml s3://compsoc/betterinformatics/courses.yaml
+      run: aws s3 --endpoint https://ams3.digitaloceanspaces.com cp --acl public-read courses.json s3://compsoc/betterinformatics/courses.json
```

Comment:
You probably also want `if: "steps.generate_yml.outputs.has_changed == 'true'"` here